### PR TITLE
Implement JettyRPCFramework on ee10 for Jetty 12

### PR DESCRIPTION
**Pull request by ddworak**

### CODE REVIEW

File: jetty/src/main/scala/com/avsystem/commons/jetty/rpc/JettyRPCFramework.scala [hunk 1]
```diff
@@ -89,30 +88,38 @@ object JettyRPCFramework extends StandardRPCFramework with LazyLogging {
...
+          completeWith(Future.fromTry(call).flatMapNow(handlePost).andThenNow {
             case Success(responseContent) =>
               response.setContentType(MimeTypes.Type.APPLICATION_JSON.asString())
               response.setCharacterEncoding(StandardCharsets.UTF_8.name())
               response.getWriter.write(responseContent.s)
             case Failure(t) =>
               response.sendError(HttpStatus.INTERNAL_SERVER_ERROR_500, t.getMessage)
               logger.error("Failed to handle RPC call", t)
-          }.andThenNow { case _ => async.complete() }
+          })
```
Comment (sebaciv) [inline jetty/src/main/scala/com/avsystem/commons/jetty/rpc/JettyRPCFramework.scala:120]: That does not seem right to me, `asyncContext.complete()` is not called after `Future`. Argument `code` in `completeWith` method is a Future, so:
```
code // some future running in background
asyncContext.complete() // imidietly closes context without waiting for actual completion

```

Comment (sebaciv) [inline jetty/src/main/scala/com/avsystem/commons/jetty/rpc/JettyRPCFramework.scala:120]: That does not seem right to me, `asyncContext.complete()` is not called after `Future` finishes. Argument `code` in `completeWith` method is a Future, so:
```
def completeWith(code: => Unit): Unit = {
  code // some future running in background
  asyncContext.complete() // immediately closes context without waiting for actual completion
}
```

**Review by sebaciv — CHANGES_REQUESTED** [review]

File: jetty/src/main/scala/com/avsystem/commons/jetty/rpc/JettyRPCFramework.scala [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
+import org.eclipse.jetty.ee10.servlet.ServletContextHandler
 import org.eclipse.jetty.http.{HttpMethod, HttpStatus, MimeTypes}
-import org.eclipse.jetty.server.handler.AbstractHandler
-import org.eclipse.jetty.server.{Handler, Request}
+import org.eclipse.jetty.server.Handler
 
-import scala.concurrent.duration._
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.atomic.AtomicBoolean
+import scala.concurrent.duration.*
```
Comment (sebaciv) [inline jetty/src/main/scala/com/avsystem/commons/jetty/rpc/JettyRPCFramework.scala:16]: Nit: switch to Scala 3 syntax? IMO we should do it in all classes or keep using Scala 2 _

File: jetty/src/main/scala/com/avsystem/commons/jetty/rpc/JettyRPCFramework.scala (hunk 1 above)
Comment (ddworak) [reply jetty/src/main/scala/com/avsystem/commons/jetty/rpc/JettyRPCFramework.scala]: Yup, copied that from udash-rest without checking yet. Should be better now if I understand the intention correctly.

**Review by sebaciv — APPROVED** [review]

**Pull request merged**